### PR TITLE
Migrate global components

### LINKED ISSUE #193: Migrate global components
Mainly `Nav` and `Footer` (and any other components they depend on)

### PR DESCRIPTION
**Pull request by jfidiles**
#### Description
Me and @peduarte added some global components to migrate the pages more easily.
We've added: 
- Icons
- Footer component
- ContactCard component
- Nav component
- NavLink component
- layout improvements

Also we removed the associated jekyll files.

#### Issues closing 
Closes #193

### CODE REVIEW

**Comment by peduarte** [comment]
Deploy preview for *fathomlondon* ready!

Built with commit b05cf45c722b5cceff0eeecbf75591c062a219ac

https://deploy-preview-195--fathomlondon.netlify.com

File: src/components/ContactCard/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export function ContactCard({ email, tel, className }) {
+	return (
+		<div className={`flex-m ${className}`}>
+			<address className="mb1 mb0-m mr3-m">
+				Irongate House, 30 Dukes Place<br />
+				London, EC3A 7LP
+			</address>
+			<p>
+				<a href={`mailto: ${email}`} className="hover-white">
```
Comment (artisologic) [inline src/components/ContactCard/index.js]: not sure the space after : will work

File: src/components/ContactCard/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+					{email}
+				</a>
+				<br />
+				<a href={`tel: ${tel}`} className="hover-white">
```
Comment (artisologic) [inline src/components/ContactCard/index.js]: not sure the space after : will work

File: src/components/Footer/index.js [hunk 3]
```diff
@@ -1,4 +1,75 @@
...
+				<NavLinks />
+			</nav>
+		</div>
+	);
+}
+
+/**
+ * TODO: Add javascript back
+ *
+ * <footer class="pv3 pv5-m bg-black white">
```
Comment (artisologic) [inline src/components/Footer/index.js]: Seems you missed the inclusion of the contact card down here? Or is this the old code still here? Maybe only keep the non migrated part?

File: src/components/ContactCard/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+export function ContactCard({ email, tel, className }) {
+	return (
+		<div className={`flex-m ${className}`}>
```
Comment (artisologic) [inline src/components/ContactCard/index.js]: Do we want to use the `classnames` package for things like that?

File: src/components/ContactCard/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+export function ContactCard({ email, tel, className }) {
```
Comment (artisologic) [inline src/components/ContactCard/index.js]: I can't see the old contact card liquid file being removed

File: src/components/Footer/index.js (hunk 3 above)
Comment (jfidiles) [reply src/components/Footer/index.js]: Good spot!

File: src/components/ContactCard/index.js (hunk 4 above)
Comment (jfidiles) [reply src/components/ContactCard/index.js]: At this point, I think yes. We can improve it later on and add the class directly in this file if the ContactCard is not reused

File: _includes/contact-card.html [hunk 6]
```diff
@@ -1,10 +0,0 @@
-<div class="flex-m {{ include.extraClass }}">
```
Comment (jfidiles) [inline _includes/contact-card.html]: @artisologic the `contact-card` liquid file it's removed here

File: src/components/NavLinks/index.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	];
+
+	// TODO: Remove this wrapping div when we move to React 16
+	// TODO: We have to add the active class later
+	return (
+		<div className="flex flex-column db-m">
+			{links.map(link => (
+				<a key={link.name} href={link.path}>
+					{link.name}
+				</a>
```
Comment (peduarte) [inline src/components/NavLinks/index.js]: I think we should add in `gatsby-link` and start using it whenever we're linking to other internal pages.

File: src/components/Footer/index.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+<script type="text/javascript" src="{{ "/assets/js/es6-support.js" | relative_url }}"></script>
+<script>
+  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
+
+  ga('create', 'UA-70512118-1', 'auto');
+  ga('send', 'pageview');
+</script>
```
Comment (jjenzz) [inline src/components/Footer/index.js]: Netlify can do this for us:

image

Comment (peduarte) [reply src/components/Footer/index.js]: That's cool 😎

Do you think it's "better" to do it via Netlify @jjenzz or have it in the `Footer` or something similar?

The only benefit I'm seeing with injecting those via Netlify, is so a person without access to the source code can add/edit scripts. Which to be fair, sounds like a good idea.

**Comment by jjenzz** [comment]
@jfidiles Approved as long as @peduarte's comment is addressed regarding `gatsby-link` 🙂

File: src/components/Footer/index.js (hunk 8 above)
Comment (jjenzz) [reply src/components/Footer/index.js]: @peduarte I've not actually used this feature before, just know it's there and been wanting to try it out. 

Yeah, it seems it allows more control over the snippets and I'm presuming it only injects them into the `master` build so our development clicks don't get recorded 🤷‍♀️ definitely assumption though, would need to test.

File: src/components/NavLinks/index.js (hunk 7 above)
Comment (jfidiles) [reply src/components/NavLinks/index.js]: Cool 👍

File: src/components/Footer/index.js (hunk 8 above)
Comment (peduarte) [reply src/components/Footer/index.js]: Ohhh that's cool! I'd be up for giving it a try 👍

**Comment by jfidiles** [comment]
Added and used `gatsby-link`

File: src/pages/index.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 
 function Index() {
```
Comment (AndrewJDick) [inline src/pages/index.js]: Could we not just export default function here?

Comment (lukemartin) [reply src/pages/index.js]: default exports are the devil. I think all exports should be explicit:

`export { Index };`

@peduarte thoughts?

Comment (jfidiles) [reply src/pages/index.js]: We can but I think @peduarte prefers to have the main pages exported as is and components to be named exports.

Comment (peduarte) [reply src/pages/index.js]: I also think we should only do named exports, but I thought Gatsby requires Page Components to be default exports... can you double check @jfidiles ?

**Comment by peduarte** [comment]
Jimmy can you add the `Closes` keyword to the description of this PR, so when it gets merged, it automatically closes https://github.com/fathomlondon/fathomlondon.com/issues/193

**Comment by jfidiles** [comment]
The `Page` coomponents require to be default exports. Haven't found anything in documentation but did found this while I changed one of the files: 

screen shot 2018-02-08 at 14 13 46

**Pull request merged**